homeassistant/components/mqtt/__init__.py

```diff
@@ -683,6 +684,40 @@ def remove():
         DOMAIN, SERVICE_PUBLISH, async_publish_service, schema=MQTT_PUBLISH_SCHEMA
     )
 
+    async def async_dump_service(call: ServiceCall):
+        """Handle MQTT dump service calls."""
+        messages = []
```

Comment:
Maybe the nonblock deque from collection they can limit the size and work as a ring buffer? So you have a third service option to limit the amount of messages?

Reply:
Since it's for debugging, we really want to have all the messages.